boards/nrf51dk/Makefile

```diff
@@ -30,5 +32,39 @@ clean:
 .PHONY: debug
 debug: target/nrf51/debug/nrf51dk
 
-#TODO: Figure out how to program the nRF51dk
+.PHONY: apps/$(APP)/build/$(TOCK_ARCH)/app.bin
+apps/$(APP)/build/$(TOCK_ARCH)/app.bin:
+	@make -C apps/$(APP) TOCK_ARCH=$(TOCK_ARCH)
+
+target/nrf51/release/nrf51dk-$(APP): target/nrf51/release/nrf51dk apps/$(APP)/build/$(TOCK_ARCH)/app.bin
+	@$(OBJCOPY) --update-section .apps=../../userland/examples/$(APP)/build/$(TOCK_ARCH)/app.bin \
+	  --set-section-flags .apps=alloc,code \
+	  target/nrf51/release/nrf51dk $@
+
+target/nrf51/release/nrf51dk.hex: target/nrf51/release/nrf51dk
+	@$(OBJCOPY) -Oihex $^ $@
+
+target/nrf51/release/nrf51dk-$(APP).hex: target/nrf51/release/nrf51dk-$(APP)
+	@$(OBJCOPY) -Oihex $^ $@
+
+JLINK=JLinkExe
+JLINK_OPTIONS+=-device nrf51422 -if swd -speed 1200 -AutoConnect 1
+JLINK_SCRIPTS_DIR=jtag/
+
+# Upload the kernel over JTAG
+.PHONY: program
+program: target/nrf51/release/nrf51dk.hex
+	$(JLINK) $(JLINK_OPTIONS) $(JLINK_SCRIPTS_DIR)/flash-kernel.jlink
+
+# Upload kernel + app over JTAG
+.PHONY: program-full
+program-full: target/nrf51/release/nrf51dk-$(APP).hex
```

Comment:
Testing and debugging. 

As I'm hacking on the kernel, I'd like to have a simple test app that shows the kernel boot sequence (including processes) is still operating properly. In practice, I've used blink, but once the UART is up and running I'd use a user-space console of some kind.

Having both in a single hex file can make it simpler to track down problems. Specifically, I can see the entire image and its layout in memory, rather than having to consder where/how things have been dynamically installed.